azuredevops.go

```diff
@@ -90,179 +101,52 @@ func runAzureDevOpsHandler(c *gin.Context) {
 		AuthHeader: c.GetHeader("Authorization"),
 	}
 
-	i := importBody{}
-	err := c.BindJSON(&i)
+	i, err := bindImportDetails(c)
 	if err != nil {
-		c.Error(err)
-		c.JSON(http.StatusBadRequest, err.Error())
+		ginutil.WriteInvalidBindError(c, err,
+			"One or more parameters failed to parse when reading the request body for import details.")
 		return
 	}
-
 	fmt.Println("from json: ", i)
 
 	if i.Group == "" {
 		fmt.Println("Unable to get due to empty group.")
-		c.JSON(http.StatusBadRequest, "Unable to get due to empty group.")
+		ginutil.WriteInvalidParamError(c, err, "group",
+			"Unable to import due to empty group.")
```

Comment:
`err` is `nil` here. As we don't have a `ginutil.WriteInvalidParam` without the Error suffix, you could create your own error here if you want. Such as:

```suggestion
		err := errors.New("missing required property: group")
		ginutil.WriteInvalidParamError(c, err, "group",
			"Unable to import due to empty group.")
```